common/lib/src/realm_types.dart

```diff
@@ -105,6 +105,15 @@ class Decimal128 {} // TODO Support decimal128 datatype https://github.com/realm
 /// @nodoc
 abstract class RealmObjectMarker {}
 
+/// @nodoc
+abstract class TopLevelObjectMarker implements RealmObjectMarker {}
+
+/// @nodoc
+abstract class EmbeddedObjectMarker implements RealmObjectMarker {}
+
+/// @nodoc
+abstract class AsymmetricObjectMarker implements RealmObjectMarker {}
```

Comment:
If this is not used I suggest not to introduce it here. or comment it out. Generally anticipating code changes is not good practice imo, since we might end up not needing that when we get to implementing it. This was the case with `RealmObjectBaseMarker`. 
Why not revert the change to `RealmObjectBaseMarker` since it seems we need it.

Reply:
So in a sense I want to say that I prefer the `RealmObjectBaseMarker` and `RealmIObjectMarker` than naming this `ToplLevelMarker` which is a new concept which is not clear to me.

Reply:
Okay, was just following core
```c++
enum class ObjectType : uint8_t { TopLevel = 0, Embedded = 0x1, TopLevelAsymmetric = 0x2 };
```
I will rename `RealmObjectMarker` to `RealmObjectBaseMarker` and `TopLevelObjectMarker` as `RealmObjectMarker`